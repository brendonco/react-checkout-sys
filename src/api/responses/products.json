[{
      "id": "classic",
      "name": "Classic Ad",
      "price": "269.99",
      "sortOrder": 1,
      "specialPriceRules": [{
          "code": "3for2",
          "companyIds": [
            "unilever"
          ]
        },
        {
          "code": "5for4",
          "companyIds": [
            "ford"
          ]
        }
      ]
    },
    {
      "id": "standout",
      "name": "Standout Ad",
      "price": "322.99",
      "sortOrder": 2,
      "specialPriceRules": [{
          "code": "7forLess",
          "companyIds": [
            "apple"
          ],
          "price": "299.99"
        },
        {
          "code": "4forLess",
          "companyIds": [
            "ford"
          ],
          "price": "309.99"
        }
      ]
    },
    {
      "id": "premium",
      "name": "Premium Ad",
      "price": "394.99",
      "sortOrder": 3,
      "specialPriceRules": [{
          "code": "4orMore",
          "companyIds": [
            "nike"
          ],
          "price": "379.99"
        },
        {
          "code": "3orMore",
          "companyIds": [
            "ford"
          ],
          "price": "389.99"
        }
      ]
    }

  ]