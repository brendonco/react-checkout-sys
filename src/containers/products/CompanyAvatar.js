import React from 'react';

const CompanyAvatar = ({
  match:{
    params: { companyId = "default" } = {}
  }
}) => {
  return <img className="company__avatar" src={require(`assets/images/${companyId}-logo.svg`)} alt='avatar' />;
};

export default CompanyAvatar;
