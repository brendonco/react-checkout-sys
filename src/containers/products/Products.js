import React from 'react';
import {
  ProductTableHeader,
  ProductTableBody
} from './widgets';

const Products = ({ products = [], productState, handleQtyChange, savedData, grandTotal}) => {
  return (
    <div>
      <ProductTableHeader />
      <ProductTableBody
        products={products}
        handleQtyChange={handleQtyChange}
      />
      <div className="row footer">
        <div className="col"></div>
        <div className="col prod__label">Grand Total:</div>
        <div className="col prod__total">{`${grandTotal}`}</div>
      </div>
    </div>
  );
}

export default Products;