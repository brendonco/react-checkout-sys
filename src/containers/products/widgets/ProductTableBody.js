import React from 'react';
import { Textfield } from 'gdp-components/textfield';

const ProductTableBody = ({ products = [], handleQtyChange }) => {
  const prods = products.map((product, idx) => 
    <div key={idx} className="row">
      <div className="col col__item col__item--description">{product.name}</div>
      <div className="col col__item col__item--quantity">
        <Textfield
          type="number"
          name={product.id}
          min="1"
          pattern="[1-9]+"
          onChange={handleQtyChange}
          onBlur={handleQtyChange}
        />
      </div>
      <div className="col col__item">${product.displayPrice}</div>
    </div>
  );

  return <form>{prods}</form>;
};

export default ProductTableBody;
