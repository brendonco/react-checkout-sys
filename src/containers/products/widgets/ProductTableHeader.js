import React from 'react';

// Table header data should be from configuration api
const ProductTableHeader = () => [{
    header1: "Description",
    header2: "Quantity",
    header3: "Price"
  }].map((head, idx) =>
  <div key={idx} className="row header">
    <div className="col">{head.header1}</div>
    <div className="col">{head.header2}</div>
    <div className="col">{head.header3}</div>
  </div>
);

export default ProductTableHeader;