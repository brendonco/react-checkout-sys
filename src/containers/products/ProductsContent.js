import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchProductList, updateProducts } from 'gdp-redux/products/actions';
import { getGrandTotal } from 'gdp-redux/products/helper';
import Products from './Products';

class ProductsContent extends React.Component {
  constructor(props){
    super(props);
    this.state = {};

    this.handleQtyChange = this.handleQtyChange.bind(this);
  }
  componentWillMount() {
    this.props.fetchProductList();
  }

  handleQtyChange(e) {
    const {
      products,
      match:{
        params: { companyId } = {}
      },
      savedData
    } = this.props;

    const prodUpdate = {
      products,
      productId: e.target.name,
      qty: e.target.value,
      companyId,
      savedData
    }

    this.props.updateProducts(prodUpdate);
  }

  render() {
    const {
      isFetch,
      products,
      savedData
    } = this.props;

    return isFetch ?
      <Products
        products={products}
        handleQtyChange={this.handleQtyChange}
        savedData={savedData}
        grandTotal={getGrandTotal(savedData)}
      /> : null // loader
  }
}

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchProductList,
  updateProducts,
}, dispatch);

ProductsContent.propTypes = {
  isFetch: PropTypes.bool,
  products: PropTypes.array,
  savedData: PropTypes.array,
}

export default connect(
  state => ({
    products: state.products.data,
    savedData: state.products.checkout,
    isFetch: state.products.isFetch,
  }),
  mapDispatchToProps
)(ProductsContent);