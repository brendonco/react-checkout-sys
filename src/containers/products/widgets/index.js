import ProductTableHeader from './ProductTableHeader';
import ProductTableBody from './ProductTableBody';

export {
  ProductTableHeader,
  ProductTableBody
}
