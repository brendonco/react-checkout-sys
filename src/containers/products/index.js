// import React from 'react';
// import { push } from 'react-router-redux';
// import { bindActionCreators } from 'redux';
// import { connect } from 'react-redux';
// import { fetchProductList } from 'seek-redux/products/actions';

// const Products = props => (
//   <div>
//     <h1>Products</h1>
//     <button onClick={() => props.changePage()}>Go to checkout page via redux</button>
//     <button onClick={() => props.fetchProductList()}>Fetch Product List</button>
//   </div>
// );

// const mapDispatchToProps = dispatch => bindActionCreators({
//   fetchProductList,
//   changePage: () => push('/checkout')
// }, dispatch);

// export default connect(
//   state => ({
//     products: state.products
//   }), 
//   mapDispatchToProps
// )(Products);


import React from 'react';
import ProductsContent from './ProductsContent';
import CompanyAvatar from './CompanyAvatar';

const Products = props => (
  <div className="wrapper">
    <CompanyAvatar {...props} />
    <ProductsContent {...props} />
  </div>
);

export default Products;