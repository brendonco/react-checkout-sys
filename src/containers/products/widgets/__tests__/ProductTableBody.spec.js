import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTableBody } from 'containers/products/widgets';

describe('Render ProductsContent', () => {
  it('Renders products', () => {
    const mockProducts = [
      {
        "id": "classic",
        "name": "Classic Ad",
        "price": "269.99",
        "sortOrder": 1,
        "specialPriceRules": [
          {
            "code": "3for2",
            "companyIds": [
              "unilever"
            ]
          },
          {
            "code": "5for4",
            "companyIds": [
              "ford"
            ]
          }
        ],
        "displayPrice": "3,239.88"
      },
      {
        "id": "standout",
        "name": "Standout Ad",
        "price": "322.99",
        "sortOrder": 2,
        "specialPriceRules": [
          {
            "code": "7forLess",
            "companyIds": [
              "apple"
            ],
            "price": "299.99"
          },
          {
            "code": "4forLess",
            "companyIds": [
              "ford"
            ],
            "price": "309.99"
          }
        ],
        "displayPrice": "7,105.78"
      },
      {
        "id": "premium",
        "name": "Premium Ad",
        "price": "394.99",
        "sortOrder": 3,
        "specialPriceRules": [
          {
            "code": "4orMore",
            "companyIds": [
              "nike"
            ],
            "price": "379.99"
          },
          {
            "code": "3orMore",
            "companyIds": [
              "ford"
            ],
            "price": "389.99"
          }
        ],
        "displayPrice": "126,916.66"
      }
    ]
    
    const component = renderer.create(<ProductTableBody products={mockProducts} handleQtyChange={(e) => {}}/>);

    expect(component).toMatchSnapshot();
  })
})