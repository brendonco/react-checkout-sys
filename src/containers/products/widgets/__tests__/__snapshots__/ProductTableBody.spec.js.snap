// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProductsContent Renders products 1`] = `
<form>
  <div
    className="row"
  >
    <div
      className="col col__item col__item--description"
    >
      Classic Ad
    </div>
    <div
      className="col col__item col__item--quantity"
    >
      <input
        className="textfield"
        min="1"
        name="classic"
        onBlur={[Function]}
        onChange={[Function]}
        pattern="[1-9]+"
        type="number"
      />
    </div>
    <div
      className="col col__item"
    >
      $
      3,239.88
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col__item col__item--description"
    >
      Standout Ad
    </div>
    <div
      className="col col__item col__item--quantity"
    >
      <input
        className="textfield"
        min="1"
        name="standout"
        onBlur={[Function]}
        onChange={[Function]}
        pattern="[1-9]+"
        type="number"
      />
    </div>
    <div
      className="col col__item"
    >
      $
      7,105.78
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col__item col__item--description"
    >
      Premium Ad
    </div>
    <div
      className="col col__item col__item--quantity"
    >
      <input
        className="textfield"
        min="1"
        name="premium"
        onBlur={[Function]}
        onChange={[Function]}
        pattern="[1-9]+"
        type="number"
      />
    </div>
    <div
      className="col col__item"
    >
      $
      126,916.66
    </div>
  </div>
</form>
`;
